ansible/library/conn_graph_facts.py

```diff
@@ -338,7 +466,25 @@ def main():
             if host_vlan:
                 device_vlan_range.append(host_vlan["VlanRange"])
                 device_vlan_list.append(host_vlan["VlanList"])
-                device_vlan_map_list[hostname] = host_vlan["VlanList"]
+                port_vlans = lab_graph.get_host_port_vlans(hostname)
+                device_vlan_map_list[hostname] = {}
+
+                port_name_list_sorted = get_port_name_list(dev['HwSku'])
+
+                for a_host_vlan in host_vlan["VlanList"]:
+                    # Get the corresponding port for this vlan from the port vlan list for this hostname
+                    found_port_for_vlan = False
+                    for a_port in port_vlans:
```

Comment:
Do we really need to check the vlan ids in `VlanList` is in `port_vlans`? `VlanList`  comes from the combination of all the vlan lists defined in `self.vlanport[hostname]['vlanlist']`, while `port_vlans` is just `self.vlanport`?

Reply:
I think it is safe to assume that self.vlanport[hostname]['vlanlist'] would always be a list of 1.  Can you please confirm? If so, then I can change the code appropriately.